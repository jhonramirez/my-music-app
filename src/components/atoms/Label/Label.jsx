import "./Label.scss";

export const Label = ({ name, className }) => {
  return <label className={className}>{name}</label>;
};
