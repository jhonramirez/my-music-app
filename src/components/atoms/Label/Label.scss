.label {
  color: rgb(240, 150, 118);
  padding: 0.5rem;
}
