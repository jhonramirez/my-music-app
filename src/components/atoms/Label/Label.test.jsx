import { render, screen } from "@testing-library/react";
import { Label } from "./Label";
import "@testing-library/jest-dom";

describe("Pruebas en <Label />", () => {
  const { container } = render(
    <Label name={"proof"} className="className" htmlFor={"htmlFor"} />
  );
  const LabelElement = screen.getByText("proof");

  test("Render Button y match con el snapshot", () => {
    expect(LabelElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("Se evalua que tenga un className", () => {
    expect(LabelElement).toHaveClass("className");
  });

  test("Se evalua que tenga un htmlFor", () => {
    expect(LabelElement).toHaveAttribute("for", "htmlFor");
  });
});
