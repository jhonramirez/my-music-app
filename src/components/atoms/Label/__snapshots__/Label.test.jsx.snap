// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Label /> Render Button y match con el snapshot 1`] = `
<div>
  <label
    class="className"
    data-testid="label"
    for="htmlFor"
  >
    proof
  </label>
</div>
`;
